### PR TITLE
Reject expression and Unit tests

### PR DESCRIPTION
**Pull request by sbezverk**
This PR add support for Reject expression and Unit tests.

@stapelberg Appreciate if you could review this one and the one before at your earliest convenience. Thanks a lot for your understanding..

### CODE REVIEW

File: nftables_test.go [hunk 1]
```diff
@@ -2541,3 +2541,118 @@ func TestMasq(t *testing.T) {
...
+				[]byte("\x02\x00\x00\x00\x0b\x00\x01\x00\x66\x69\x6c\x74\x65\x72\x00\x00\x08\x00\x02\x00\x00\x00\x00\x00"),
+				// nft add chain ip filter base-chain
+				[]byte("\x02\x00\x00\x00\x0b\x00\x01\x00\x66\x69\x6c\x74\x65\x72\x00\x00\x0f\x00\x03\x00\x62\x61\x73\x65\x2d\x63\x68\x61\x69\x6e\x00\x00"),
+				// nft add  rule ip filter base-chain reject with tcp reset
+				[]byte("\x02\x00\x00\x00\x0b\x00\x01\x00\x66\x69\x6c\x74\x65\x72\x00\x00\x0f\x00\x02\x00\x62\x61\x73\x65\x2d\x63\x68\x61\x69\x6e\x00\x00\x28\x00\x04\x80\x24\x00\x01\x80\x0b\x00\x01\x00\x72\x65\x6a\x65\x63\x74\x00\x00\x14\x00\x02\x80\x08\x00\x01\x00\x00\x00\x00\x01\x05\x00\x02\x00\x01\x00\x00\x00"),
+				// batch end
+				[]byte("\x00\x00\x00\x0a"),
+			},
+			rejectExprs: []expr.Any{
+				&expr.Reject{Type: 0x1, Code: unix.NFT_REJECT_TCP_RST},
```
Comment (stapelberg) [inline nftables_test.go]: Use `unix.NFT_REJECT_TCP_RST` instead of 0x1 here.

File: nftables_test.go [hunk 2]
```diff
@@ -2541,3 +2541,118 @@ func TestMasq(t *testing.T) {
...
+				[]byte("\x02\x00\x00\x00\x0b\x00\x01\x00\x66\x69\x6c\x74\x65\x72\x00\x00\x08\x00\x02\x00\x00\x00\x00\x00"),
+				// nft add chain ip filter base-chain
+				[]byte("\x02\x00\x00\x00\x0b\x00\x01\x00\x66\x69\x6c\x74\x65\x72\x00\x00\x0f\x00\x03\x00\x62\x61\x73\x65\x2d\x63\x68\x61\x69\x6e\x00\x00"),
+				// nft add  rule ip filter base-chain reject with icmp type host-unreachable
+				[]byte("\x02\x00\x00\x00\x0b\x00\x01\x00\x66\x69\x6c\x74\x65\x72\x00\x00\x0f\x00\x02\x00\x62\x61\x73\x65\x2d\x63\x68\x61\x69\x6e\x00\x00\x28\x00\x04\x80\x24\x00\x01\x80\x0b\x00\x01\x00\x72\x65\x6a\x65\x63\x74\x00\x00\x14\x00\x02\x80\x08\x00\x01\x00\x00\x00\x00\x00\x05\x00\x02\x00\x00\x00\x00\x00"),
+				// batch end
+				[]byte("\x00\x00\x00\x0a"),
+			},
+			rejectExprs: []expr.Any{
+				&expr.Reject{Type: 0x0, Code: unix.NFT_REJECT_ICMP_UNREACH},
```
Comment (stapelberg) [inline nftables_test.go]: Use `unix.NFT_REJECT_ICMP_UNREACH` instead of 0x0 here.

File: nftables_test.go (hunk 1 above)
Comment (sbezverk) [reply nftables_test.go]: done

File: nftables_test.go (hunk 2 above)
Comment (sbezverk) [reply nftables_test.go]: done

**Pull request merged**